lib/hon_graffiti_phoenix/parsers/markup_parser.ex

```diff
@@ -0,0 +1,64 @@
+defmodule HonGraffitiPhoenix.Parsers.MarkupParser do
+  alias HonGraffitiPhoenix.Parsers.DecoratedString
+
+  @color_map %{
+    "w" => "white", "*" => "white", "r" => "red",
+    "b" => "blue", "y" => "yellow", "m" => "magenta",
+    "n" => "brown", "p" => "purple", "o" => "orange",
+    "t" => "teal", "v" => "grey", "g" => "green",
+    "k" => "black", "" => "black"}
+
+  @moduledoc """
+  Parses strings using standard HoN style markup.
+  """
+  @type meta_string :: %DecoratedString{color: String.t, body: String.t}
+
+  # This captures(named style) on a caret followed by EITHER
+  # one of the characters, or 3 digits
+  @capture_style_separately ~r/\^(?<color>\d{3}|[wrbymnpkotvg*])(?<body>.*)/i
+  @caret_regex ~r/\^.*?(?=\^)/
+
+#  Public functions
+  @spec parse :: []
+  def parse, do: []
+
+  @spec parse(String.t) :: [DecoratedString]
+  def parse(quote) when quote == nil, do: []
+  def parse(quote) do
+    quote
+    |> String.split(@caret_regex, trim: true, include_captures: true)
+    |> Enum.map(fn segment -> parse_segment(segment) end)
+  end
+
+  @doc """
+  Separates a String into it's markup and body.
+  If no markup is found, a default is given.
+  """
+  @spec parse_segment(String.t) :: meta_string
+  def parse_segment(string) do
```

Comment:
Awesome, IMO that's the ideal 👍  Get dem interfaces expressing intent, nam sayin?